modules/components/Link.js

```diff
@@ -67,14 +67,25 @@ var Link = React.createClass({
     event.preventDefault();
 
     if (allowTransition)
-      this.transitionTo(this.props.to, this.props.params, this.props.query);
+      this.transitionTo(this.getTo(), this.props.params, this.props.query);
+  },
+
+  /**
+   * Returns the value of the 'to' prop if specified, otherwise returns the
+   * name of the currently active route.
+   */
+  getTo: function() {
+    if (this.props.to)
+      return this.props.to;
+    var currentRoutes = this.context.getCurrentRoutes()
+    return currentRoutes[currentRoutes.length - 1].name
```

Comment:
routes aren't guaranteed to have a name, would be safer to link to the current path which can be found on `this.context.router.getPath()` I think